lib/src/BatchRequestContent.d.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * @interface
+ * Signature representing BatchRequestStep data
+ * @property {string} id - Unique identity for the request, Should not be an empty string
+ * @property {string[]} [dependsOn] - Array of dependencies
+ * @property {Request} request - The Request object
+ */
+export interface BatchRequestStep {
+    id: string;
+    dependsOn?: string[];
+    request: Request;
+}
+/**
+ * @interface
+ * Signature representing single request in a Batching
+ * @extends RequestInit
+ * @see {@link https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L1337} and {@link https://fetch.spec.whatwg.org/#requestinit}
+ *
+ * @property {string} url - The url value of the request
+ */
+export interface RequestData extends RequestInit {
+    url: string;
+}
+/**
+ * @interface
+ * Signature representing batch request data
+ * @property {string} id - Unique identity for the request, Should not be an empty string
+ * @property {string[]} [dependsOn] - Array of dependencies
+ */
+export interface BatchRequestData extends RequestData {
+    id: string;
+    dependsOn?: string[];
+}
+/**
+ * @interface
+ * Signature representing batch request body
+ * @property {BatchRequestData[]} requests - Array of request data, a json representation of requests for batch
+ */
+export interface BatchRequestBody {
+    requests: BatchRequestData[];
+}
+/**
+ * Class for handling BatchRequestContent
+ */
+export declare class BatchRequestContent {
+    /**
+     * @private
+     * @static
+     * Limit for number of requests {@link - https://developer.microsoft.com/en-us/graph/docs/concepts/known_issues#json-batching}
+     */
+    private static requestLimit;
+    /**
+     * To keep track of requests, key will be id of the request and value will be the request json
+     */
+    requests: Map<string, BatchRequestStep>;
+    /**
+     * Constructs a BatchRequestContent instance
+     * @param {BatchRequestStep[]} [requests] - Array of requests value
+     */
+    constructor(requests?: BatchRequestStep[]);
+    /**
+     * Adds a request to the batch request content
+     * @param {BatchRequestStep} request - The request value
+     * @return The id of the added request
+     */
+    addRequest(request: BatchRequestStep): string;
```

Comment:
Do we need to return the id of the added Request? Could it be a 'void' method..

Reply:
Instead of returning nothing, returning id gives the sense that the request is been added. This is why we are returning id.